tcp_congestion/src/App.js

```diff
@@ -1,109 +1,225 @@
-        // lost: []
-      }
-      else {
-          setCwnd(1);
-          setSsthresh(ssthresh/2);
-          setSent(Array.from({ length: cwnd + 1 }, (v, i) => ack+1 + i));
-          // cwnd: 1,
-          // ssthresh: state.ssthresh/2,
-          // sent: Array.from({ length: this.state.cwnd + 1 }, (v, i) => this.state.ack + i)
-      }
+    if (cwnd < ssthresh && lost.length === 0) {
+      setCwnd(cwnd + 1);
+      setSent(Array.from({ length: cwnd + 1 }, (v, i) => ack + 1 + i));
+    } else if (lost.length !== 0) {
+      let last = sent[sent.length - 1];
+      setCwnd(cwnd + 1);
+      setSent(lost);
+      if (sent.length <= cwnd)
+        setSent(
+          sent.concat(
+            Array.from(
+              { length: sent.length - cwnd + 1 },
+              (v, i) => last + 1 + i
+            )
+          )
+        );
+      setLost([]);
+    } else {
+      setCwnd(1);
+      setSsthresh(ssthresh / 2);
+      setSent(Array.from({ length: cwnd + 1 }, (v, i) => ack + 1 + i));
+    }
   };
+
+  const startSimulation = () => {
+    setCwnd(1);
+    setSsthresh(64);
+    setSent([0]);
+    setLost([]);
+    setAck(0);
+    setSimulationRunning(true);
+  };
+
+  const stopSimulation = () => {
+    setSimulationRunning(false);
+  };
+
+  // Automated simulation logic
+  useEffect(() => {
+    let interval;
+
+    if (simulationRunning) {
+      interval = setInterval(() => {
+        const randomNum = Math.random();
+
+        // Simulate packet loss
+        if (randomNum < packetLossRate / 100) {
+          setLost((prev) => [...prev, sent[sent.length - 1]]);
+        }
+
+        // Simulate acknowledgment step-by-step
+        handleClick();
+
+        // Handle next simulation step
+        handleNext();
+      }, latency); // Use latency as the interval time for slowing down the process
+
+      // Cleanup on component unmount or when simulation stops
+      return () => clearInterval(interval);
+    }
+  }, [simulationRunning, latency, packetLossRate, sent]);
+
   return (
     <div className="App">
-      <header className="App-header">
-      Simulate Congestion Control Algorithms
-      </header>
+      <header className="App-header">Congestion Control Simulator</header>
       <div className="App-body">
-      <div className="initialvalues">
-        <div className='pkthead'>
-        <h2>Congestion Window</h2>
-        { cwnd }
+        <div className="initialvalues">
+          <div className="pkthead">
+            <h2>Congestion Window</h2>
+            <p>{cwnd}</p>
+          </div>
+          <div className="pkthead">
+            <h2>Slow Start Threshold</h2>
+            <p>{ssthresh}</p>
+          </div>
+          <div className="pkthead">
+            <h3>Change Initial Slow Start Threshold</h3>
+            <input
+              type="number"
+              onChange={handleChangeThresh}
+              value={ssthresh}
+            />
+          </div>
         </div>
-        <div className='pkthead'>
```

Comment:
Take the network conditions initially, including slow start threshold (ssthresh), array of lost packets and rounds of simulation. After that the user should need to press only two buttons, start simulation and end simulation, and the simulator will automatically simulate these rounds according to the algorithm, showing which packets are being transferred in the current window.